Library/Homebrew/cmd/audit.rb

```diff
@@ -827,12 +827,26 @@ def audit_version
   def audit_checksum
     return unless checksum
 
-    case checksum.hash_type
-    when :md5
-      problem "MD5 checksums are deprecated, please use SHA1 or SHA256"
-      return
-    when :sha1   then len = 40
-    when :sha256 then len = 64
+    strict = ARGV.include? "--strict"
+
+    if strict
+      case checksum.hash_type
+      when :md5
+        problem "MD5 checksums are deprecated, please use SHA256"
+        return
+      when :sha1
+        problem "SHA1 checksums are deprecated, please use SHA256"
+        return
+      when :sha256 then len = 64
+      end
+    else
+      case checksum.hash_type
+      when :md5
+        problem "MD5 checksums are deprecated, please use SHA256"
```

Comment:
> if and elsif.

Of course. Duh Dom :smile_cat:.

> If package authors publish sha1 checksums should we use them?

IMO, this is a decent opportunity to go to upstream developers *"MacPorts and Homebrew on OS X primarily use SHA256 checksumming now on recent distros, and Ubuntu and Debian support it on recent distros as well. Please publish SHA256sums alongside/instead of SHA1".*